fda/lfd.py

```diff
@@ -0,0 +1,91 @@
+import numpy as np
+
+from fda.basis import FDataBasis, Constant, _same_domain
+
+__author__ = "Pablo Pérez Manso"
+__email__ = "92manso@gmail.com"
+
+
+class Lfd:
+    """Defines the structure of a linear differential operator function system
+
+    .. math::
+        Lx(t) = b_0(t) x(t) + b_1(t) x'(x) + \dots + b_{n-1}(t) d^{n-1}(x(t)) + b_n(t) d^n(x(t))
+
+    Attributes:
+        nderiv (int): the order of the operator. It's the n coefficient in the equation above.
+
+        bwtlist (list):  A FDataBasis objects list of length either nderiv
+
+    """
+
+    def __init__(self, param=0, domain_range=(0, 1)):
+        """Lfd Constructor
+
+        Args:
+            param (int, list, FDataBasis optional): If the given parameter is an integer, it'll be the order of
+                the highest derivative.
+                  If its a list, of integers or FDataBasis objects, it will be taken as the weight functions
+                  for the derivatives.
+                  If its a FDataBasis object, each sample will be a weight function and nderiv will be the number
+                  of samples
+
+            domain_range (tuple or list of tuples, optional): Definition of the
+                interval where the weight functions are defined. Defaults to (0,1).
+        """
+
+        if isinstance(param, list):
```

Comment:
It is probably better if we iterate over, and do not assume a list. It should also do the right think given an FDataBasis.